object_detection/utils/box_label_manager.py

```diff
@@ -0,0 +1,195 @@
+
+                """
+
+        # Add the bounding box, for fun ;-)
+        self.bounding_boxes.append(_bounding_box)
+
+        left   = _bounding_box[0]
+        right  = _bounding_box[2]
+        top    = _bounding_box[1]
+        bottom = _bounding_box[3]
+
+        # Keep the initial propose
+        first_proposal = [ left , top - _label_height, left + _label_width , top]
+
+        # The offset will slide the box propose to the right or to the left
+        x_offset = 0
+        stay_in_the_loop =  True
+        while stay_in_the_loop:
+
+            #
+            # Proposal 1
+            #
+            # First label box proposal, top/left
+
+            proposal_label_box =  [ left + x_offset, top - _label_height, left + _label_width + x_offset, top]
+
+
+            # Surface if over an existing label box 30% of area
+            if self.isOverALabel(proposal_label_box)< MAX_COVERING:
+                # We are done
+                self.label_bounding_boxes.append(proposal_label_box)
+                return proposal_label_box
+
+            #
+            # Proposal 2
+            #
+            # Second proposal, bottom/left
+            proposal_label_box = [left + x_offset, bottom , left + _label_width + x_offset, bottom + _label_height]
+            if self.isOverALabel(proposal_label_box) < MAX_COVERING:
+                # We are done
+                self.label_bounding_boxes.append(proposal_label_box)
+                return proposal_label_box
+
+            #
+            # Proposal 3
+            #
+            # Third proposal, top/right
+            proposal_label_box = [right-_label_width - x_offset, top - _label_height, right - x_offset, top]
+            if self.isOverALabel(proposal_label_box) < MAX_COVERING:
+                # We are done
+                self.label_bounding_boxes.append(proposal_label_box)
+                return proposal_label_box
+
+            #
+            # Proposal 4
+            #
+            # Third proposal, bottom/right
+            proposal_label_box = [right-_label_width - x_offset, bottom , right - x_offset, bottom + _label_height]
+            if self.isOverALabel(proposal_label_box) < MAX_COVERING:
+                # We are done
+                self.label_bounding_boxes.append(proposal_label_box)
+                return proposal_label_box
+
+            x_offset = x_offset + 20
+            if x_offset < (right - left) - _label_width:
+                stay_in_the_loop = False
+
+        # End of the while
+
+        # Last chance
+        # Rectangle is as height as the image (95%)
+        if (bottom - top) > (0.95 * self.image_height):
+            # Inside the rectangle
+            first_proposal = [left, top, left + _label_width, top + _label_height]
+
+        return first_proposal
+
+    # Does the box proposal over an existing label ?
+    def isOverALabel(self, _label_box):
+        iou = 0
+        if (_label_box[0] < 0) or (_label_box[2]>self.image_width) or \
```

Comment:
Seems that here is the validation for the input parameters, but why it will return 1 (based on my understanding, it means 100% of the label overriding) if the input parameters are invalid?

Reply:
Corrected